src/provider.js

```diff
@@ -156,10 +156,7 @@
              * @returns {boolean}
              */
             this.checkValid = function(form) {
-                if (form.$valid === undefined) {
-                    return false;
-                }
-                return (form && form.$valid === true);
+                return !!form && !!form.$valid;
```

Comment:
`return form && form.$valid` will return `undefined` or `null` instead of `false`, in case of  the `form` is `undefined` or `null` respectively. I just want this function to return boolean as we expected.